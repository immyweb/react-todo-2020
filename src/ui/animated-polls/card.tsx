import * as React from 'react';

import { APCard, APHeadline, APCopy, APCardImg, APCardCopy } from './styles';

import { ICard } from './types';

export const CardScreen: React.SFC<ICard> = ({ imageMobile, imageDesktop, title, copy }) => {
  return (
    <APCard>
      <APCardImg imageMobile={imageMobile} imageDesktop={imageDesktop} />
      <APCardCopy>
        <APHeadline>{title}</APHeadline>
        <APCopy>{copy}</APCopy>
      </APCardCopy>
    </APCard>
  );
};
