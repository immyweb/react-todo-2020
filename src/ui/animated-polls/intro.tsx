import * as React from 'react';

import {
  APIntro,
  APHeadline,
  APCopy,
  APIntroCopy,
  APIntroImg,
  APIntroLink,
  APIntroHashTag,
} from './styles';

import { IIntroScreen } from './types';

export const IntroScreen: React.SFC<IIntroScreen> = ({
  imageMobile,
  imageDesktop,
  title,
  introText,
  cta,
  ctaLink,
  ctaUrl,
}) => {
  return (
    <APIntro>
      <APIntroImg imageMobile={imageMobile} imageDesktop={imageDesktop} />
      <APIntroCopy>
        <APHeadline>{title}</APHeadline>
        {introText.map((text, i) => (
          <APCopy key={i}>{text}</APCopy>
        ))}
        <APCopy>
          <APIntroLink href={ctaUrl}>
            {cta} <APIntroHashTag>{ctaLink}</APIntroHashTag>
          </APIntroLink>
        </APCopy>
      </APIntroCopy>
    </APIntro>
  );
};
