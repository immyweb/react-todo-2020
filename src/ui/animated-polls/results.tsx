import * as React from 'react';

import { ChartHorizontalBar } from './horizontal-bar-chart';
import {
  APVote,
  APVoteImgBkgnd,
  APVoteHeadline,
  APCopy,
  APIntroLink,
  APIntroHashTag,
  APVoteContent,
  APVoteImgBkgndCover,
  APVoteImgBkgndCoverGradient,
  APResultsHeadline,
  APChartHolder,
} from './styles';

import { IResultsScreen, IVoteCount, ICard } from './types';

interface IProps extends IResultsScreen {
  options: ICard[];
}

export const processData = (cards: ICard[]): IVoteCount[] => {
  let totalVoteCount = 0;
  return cards.map(
    ({
      title,
      iconImg,
      voteCount: { voteCount, noVoteCount } = { voteCount: 0, noVoteCount: 0 },
    }) => {
      totalVoteCount += voteCount;
      const datum = {
        iconImg,
        voteCount,
        noVoteCount,
        player: title,
        get votePercentage(): number {
          return totalVoteCount ? Math.round((voteCount / totalVoteCount) * 100) : 0;
        },
      };
      return datum;
    },
  );
};

export const ResultsScreen: React.SFC<IProps> = ({
  bkgndImgMobile,
  bkgndImgDesktop,
  title,
  introTxt,
  options,
  ctaTxt,
  cta,
  ctaLink,
  ctaUrl,
  theme,
}) => {
  const graphData = processData(options);

  return (
    <APVote>
      <APVoteImgBkgnd imageMobile={bkgndImgMobile} imageDesktop={bkgndImgDesktop} />
      <APVoteImgBkgndCover />
      <APVoteImgBkgndCoverGradient />
      <APVoteContent>
        <APVoteHeadline>{title}</APVoteHeadline>
        <APCopy textAlign={'left'}>{introTxt}</APCopy>
        <APChartHolder>
          <ChartHorizontalBar data={graphData} theme={theme} />
        </APChartHolder>
        <APResultsHeadline>{ctaTxt}</APResultsHeadline>
        <APCopy>
          <APIntroLink href={ctaUrl}>
            {cta} <APIntroHashTag>{ctaLink}</APIntroHashTag>
          </APIntroLink>
        </APCopy>
      </APVoteContent>
    </APVote>
  );
};
