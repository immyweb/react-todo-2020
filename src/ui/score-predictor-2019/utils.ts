import { IFixture, ITeam, FixtureResultType } from './types';

export const groupFixtures = (fixtures: IFixture[], groupNumber: number): IFixture[][] => {
  const grouped = [];
  const fixturesCopy = fixtures.map(a => ({ ...a }));
  while (fixturesCopy.length > 0) {
    grouped.push(fixturesCopy.splice(0, groupNumber));
  }

  return grouped;
};

export const getBadge = (teamName: string, teams: ITeam[]): string => {
  const team = teams.find(t => t.team === teamName);
  if (team !== undefined) {
    return team.badge;
  }
  return '';
};

export const scoreConfig = {
  WIN: {
    type: FixtureResultType.Win,
    points: 3,
  },
  LOSE: {
    type: FixtureResultType.Lose,
    points: 0,
  },
  DRAW: {
    type: FixtureResultType.Draw,
    points: 1,
  },
};

export const groupBy = <T>(items: T[], prop: string) => {
  return items.reduce((grouped, item: T) => {
    const propertyValue = item[prop];

    if (!grouped[propertyValue]) {
      grouped[propertyValue] = [];
    }
    grouped[propertyValue].push(item);
    return grouped;
  }, {});
};
