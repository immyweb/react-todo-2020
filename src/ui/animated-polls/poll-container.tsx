import * as React from 'react';

import { IntroScreen } from './intro';
import { Nav } from './nav';
import { ProgressBar } from './progress-bar';
import { CardScreen } from './card';
import { VoteScreen } from './vote';
import { ResultsScreen } from './results';
import Error from '../error';
import Icon from '../icon/icon';
import { colours } from '../../styles/colours';
import { podTracker } from '../../utils/pod-tracker-utils';
import { transferTinderAPI } from '../../api-services';

import { APNavHolder, APProgressHolder, APNavIcon, APNavIntroHolder } from './styles';

import { ICard, IAnimatedPoll, IVoteCount } from './types';
import { IComponentProps } from '../../types/tracking';

interface IProps extends IAnimatedPoll, IComponentProps {}

interface IState {
  currentCardIndex: number;
  cards: ICard[];
  showIntro: boolean;
  showCards: boolean;
  showVote: boolean;
  showResults: boolean;
  hasErrored: boolean;
}

export class PollContainer extends React.Component<IProps, IState> {
  cachedWarmState = {};

  initialState = {
    currentCardIndex: 0,
    cards: [],
    showIntro: true,
    showCards: false,
    showVote: false,
    showResults: false,
    hasErrored: false,
  };

  constructor(props: IAnimatedPoll) {
    super(props);

    this.state = {
      ...this.initialState,
      cards: props.content.cards,
    };
  }

  async componentDidMount() {
    const {
      config: { targetClubId },
      content: { cards },
    } = this.props;
    const cardIds = cards.map((card: ICard) => card.id);
    try {
      const cardVotes = await transferTinderAPI.getPlayerVotes(targetClubId, cardIds);

      const cardsWithVotes = this.mergeVotes(cardVotes, cards) as ICard[];
      this.setState({ cards: cardsWithVotes });
    } catch (e) {
      this.setState({ hasErrored: true });
    }
  }

  mergeVotes = (results: IVoteCount[], cards: ICard[]) => {
    return cards.map(card => {
      const matchingResult = results.find(result => {
        return card.id === result.player;
      });
      return matchingResult ? { ...card, voteCount: matchingResult } : { ...card };
    });
  };

  startPoll = () => {
    this.setState({
      showIntro: false,
      showCards: true,
    });

    podTracker.click({
      id: this.props.id,
      description: `Generic Animated Poll started`,
    });
  };

  clickLeft = () => {
    podTracker.click({
      id: this.props.id,
      description: `Generic Animated Poll click back`,
    });

    this.setState({
      currentCardIndex: Math.max(0, this.state.currentCardIndex - 1),
    });
  };

  clickRight = () => {
    podTracker.click({
      id: this.props.id,
      description: `Generic Animated Poll click forward`,
    });

    const { currentCardIndex, cards } = this.state;
    if (currentCardIndex === cards.length - 1) {
      this.setState({
        showCards: false,
        showVote: true,
      });
    } else {
      this.setState({
        currentCardIndex: Math.min(cards.length, currentCardIndex + 1),
      });
    }
  };

  clickVote = async (votesYes: string[]) => {
    podTracker.click({
      id: this.props.id,
      description: `Generic Animated Poll click vote`,
    });

    const { targetClubId } = this.props.config;
    this.setState({
      showVote: false,
      showResults: true,
    });

    await transferTinderAPI.addPlayerVotes(targetClubId, votesYes, []);
  };

  render() {
    const { introScreen, voteScreen, resultsScreen } = this.props.content;
    const { theme } = this.props.config;
    const {
      showIntro,
      showCards,
      showVote,
      showResults,
      cards,
      currentCardIndex,
      hasErrored,
    } = this.state;

    return hasErrored ? (
      <Error />
    ) : (
      <div>
        {showIntro && (
          <div>
            <IntroScreen {...introScreen} />
            <APNavIntroHolder>
              <APNavIcon disabled={false} onClick={this.startPoll}>
                <Icon iconName={'chevron-solid-right'} iconSize={16} iconColour={colours.$white} />
              </APNavIcon>
            </APNavIntroHolder>
          </div>
        )}
        {showCards && (
          <div>
            <APProgressHolder>
              <ProgressBar options={cards} currentCardIndex={currentCardIndex} />
            </APProgressHolder>
            <CardScreen {...cards[currentCardIndex]} />
            <APNavHolder>
              <Nav
                currentCardIndex={currentCardIndex}
                onClickLeft={this.clickLeft}
                onClickRight={this.clickRight}
              />
            </APNavHolder>
          </div>
        )}
        {showVote && (
          <VoteScreen {...voteScreen} options={cards} theme={theme} clickVote={this.clickVote} />
        )}
        {showResults && <ResultsScreen {...resultsScreen} options={cards} theme={theme} />}
      </div>
    );
  }
}
