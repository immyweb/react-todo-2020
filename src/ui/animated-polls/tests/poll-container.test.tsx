import * as React from 'react';
import { shallow } from 'enzyme';

import { PollContainer } from '../poll-container';
import { IntroScreen } from '../intro';
import { CardScreen } from '../card';
import { VoteScreen } from '../vote';
import { ResultsScreen } from '../results';
import Error from '../../error';

import { podTracker } from '../../../utils/pod-tracker-utils';

jest.mock('../../../api-services', () => ({
  transferTinderAPI: {
    addPlayerVotes: jest.fn(),
  },
}));

jest.mock('../../../utils/pod-tracker-utils', () => ({
  podTracker: {
    click: jest.fn(),
  },
}));

const config = {
  theme: {
    primary: '#000000',
    secondary: '#ffffff',
  },
};

const content = {
  teaserScreen: {
    title: 'title',
    copy: 'copy',
    cta: 'cta',
    imageMobile: 'http://www.thesun.co.uk/imageMobile.jpg',
    imageDesktop: 'http://www.thesun.co.uk/imageDesktop.jpg',
  },
  introScreen: {
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png',
    imageDesktop: '',
    title: 'Jeremy Styles',
    introText: [
      'Jeremy Kyle needs help picking his moustache style.',
      'Flick through the gallery of images and vote on your favourite at the end.',
    ],
    cta: 'Read more from:',
    ctaLink: '#MOVEMBER',
    ctaUrl: 'http://www.thesun.co.uk',
  },
  cards: [
    {
      id: 'chevron',
      iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png',
      imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg',
      imageDesktop: '',
      title: 'The Chevron',
      copy: 'Looks a bit more Bruce Forsyth than Tom Selleck...',
    },
    {
      id: 'handlebar',
      iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png',
      imageMobile: '',
      imageDesktop: '',
      title: 'The Handlebar',
      copy: 'Get a grip!',
    },
    {
      id: 'pencil',
      iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png',
      imageMobile: '',
      imageDesktop: '',
      title: 'The Pencil',
      copy: 'Work it out with this',
    },
  ],
  voteScreen: {
    bkgndImgMobile: 'handlebar-mobile.jpg',
    bkgndImgDesktop: 'desktop.jpg',
    title: 'Have your say',
    introTxt: 'Which do you think looked best? Vote on your favourite moustache',
    ctaTxt: 'Tell us what you think',
    cta: 'Read more from:',
    ctaLink: '#MOVEMBER',
    ctaUrl: 'http://www.thesun.co.uk',
  },
  resultsScreen: {
    bkgndImgMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg',
    bkgndImgDesktop: '',
    title: 'The Results',
    introTxt: 'Hereâ€™s what everyone else thought, did you agree with the others?',
    ctaTxt: 'Thanks for voting!',
    cta: 'Read more from:',
    ctaLink: '#MOVEMBER',
    ctaUrl: 'http://www.thesun.co.uk',
  },
};

function shallowRender(props?: any) {
  return shallow(<PollContainer id={'card'} config={config} content={content} {...props} />);
}

describe('<PollContainer />', () => {
  it('renders correctly when errored', () => {
    const wrapper = shallowRender();
    const error = wrapper.find(Error);
    expect(error.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when not errored', () => {
    const wrapper = shallowRender();
    wrapper.setState({
      hasErrored: false,
    });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('sets correct state values on launch', () => {
    const wrapper = shallowRender();
    expect(wrapper.state('showIntro')).toEqual(true);
    expect(wrapper.state('showCards')).toEqual(false);
    expect(wrapper.state('showVote')).toEqual(false);
    expect(wrapper.state('showResults')).toEqual(false);
  });

  it('startPoll hides <IntroScreen />, shows <CardScreen /> and tracks event', () => {
    const wrapper = shallowRender();
    const instance = wrapper.instance() as PollContainer;
    wrapper.setState({
      hasErrored: false,
    });
    instance.startPoll();
    expect(wrapper.state('showIntro')).toEqual(false);
    expect(wrapper.state('showCards')).toEqual(true);
    expect(wrapper.state('showVote')).toEqual(false);
    expect(wrapper.state('showResults')).toEqual(false);
    wrapper.update();
    const introScreen = wrapper.find(IntroScreen);
    expect(introScreen.length).toEqual(0);
    const cardScreen = wrapper.find(CardScreen);
    expect(cardScreen.length).toEqual(1);

    expect(podTracker.click).toHaveBeenCalledWith({
      description: 'Generic Animated Poll started',
      id: 'card',
    });
  });

  it('clickLeft function sets correct currentCardIndex index and tracks event', () => {
    const wrapper = shallowRender();
    const instance = wrapper.instance() as PollContainer;
    wrapper.setState({
      hasErrored: false,
    });
    instance.clickLeft();
    expect(wrapper.state('currentCardIndex')).toEqual(0);
    wrapper.setState({
      currentCardIndex: 3,
    });
    instance.clickLeft();
    expect(wrapper.state('currentCardIndex')).toEqual(2);

    expect(podTracker.click).toHaveBeenCalledWith({
      description: 'Generic Animated Poll click back',
      id: 'card',
    });
  });

  it('clickRight function sets correct currentCardIndex index, correct state and tracks event', () => {
    const wrapper = shallowRender();
    const instance = wrapper.instance() as PollContainer;
    wrapper.setState({
      hasErrored: false,
    });
    instance.clickRight();
    expect(wrapper.state('currentCardIndex')).toEqual(1);
    instance.clickRight();
    expect(wrapper.state('currentCardIndex')).toEqual(2);

    expect(podTracker.click).toHaveBeenCalledWith({
      description: 'Generic Animated Poll click forward',
      id: 'card',
    });
  });

  it('mergeVotes function adds correct vote count to player', () => {
    const wrapper = shallowRender();
    const instance = wrapper.instance() as PollContainer;
    const results = [
      {
        player: 'chevron',
        voteCount: 5,
        noVoteCount: 5,
      },
      {
        player: 'handlebar',
        voteCount: 10,
        noVoteCount: 0,
      },
      {
        player: 'pencil',
        voteCount: 0,
        noVoteCount: 10,
      },
    ];
    const result = instance.mergeVotes(results, content.cards);
    expect(result).toEqual([
      {
        id: 'chevron',
        iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png',
        imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg',
        imageDesktop: '',
        title: 'The Chevron',
        copy: 'Looks a bit more Bruce Forsyth than Tom Selleck...',
        voteCount: { player: 'chevron', voteCount: 5, noVoteCount: 5 },
      },
      {
        id: 'handlebar',
        iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png',
        imageMobile: '',
        imageDesktop: '',
        title: 'The Handlebar',
        copy: 'Get a grip!',
        voteCount: { player: 'handlebar', voteCount: 10, noVoteCount: 0 },
      },
      {
        id: 'pencil',
        iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png',
        imageMobile: '',
        imageDesktop: '',
        title: 'The Pencil',
        copy: 'Work it out with this',
        voteCount: { player: 'pencil', voteCount: 0, noVoteCount: 10 },
      },
    ]);
  });

  it('clickVote update the state, track an event, and send the vote data to the api', () => {
    const wrapper = shallowRender();
    const instance = wrapper.instance() as PollContainer;
    wrapper.setState({
      hasErrored: false,
      showIntro: false,
      showCards: false,
      showVote: true,
      showResults: false,
    });
    instance.clickVote(['fumanchu']);
    expect(wrapper.state('showIntro')).toEqual(false);
    expect(wrapper.state('showCards')).toEqual(false);
    expect(wrapper.state('showVote')).toEqual(false);
    expect(wrapper.state('showResults')).toEqual(true);
    wrapper.update();
    const voteScreen = wrapper.find(VoteScreen);
    expect(voteScreen.length).toEqual(0);
    const resultsScreen = wrapper.find(ResultsScreen);
    expect(resultsScreen.length).toEqual(1);

    expect(podTracker.click).toHaveBeenCalledWith({
      description: 'Generic Animated Poll click vote',
      id: 'card',
    });
  });
});
