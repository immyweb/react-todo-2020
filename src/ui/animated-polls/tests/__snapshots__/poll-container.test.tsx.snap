// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PollContainer /> renders correctly when errored 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PollContainer
    config={
      Object {
        "theme": Object {
          "primary": "#000000",
          "secondary": "#ffffff",
        },
      }
    }
    content={
      Object {
        "cards": Array [
          Object {
            "copy": "Looks a bit more Bruce Forsyth than Tom Selleck...",
            "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png",
            "id": "chevron",
            "imageDesktop": "",
            "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg",
            "title": "The Chevron",
          },
          Object {
            "copy": "Get a grip!",
            "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png",
            "id": "handlebar",
            "imageDesktop": "",
            "imageMobile": "",
            "title": "The Handlebar",
          },
          Object {
            "copy": "Work it out with this",
            "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png",
            "id": "pencil",
            "imageDesktop": "",
            "imageMobile": "",
            "title": "The Pencil",
          },
        ],
        "introScreen": Object {
          "cta": "Read more from:",
          "ctaLink": "#MOVEMBER",
          "ctaUrl": "http://www.thesun.co.uk",
          "imageDesktop": "",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png",
          "introText": Array [
            "Jeremy Kyle needs help picking his moustache style.",
            "Flick through the gallery of images and vote on your favourite at the end.",
          ],
          "title": "Jeremy Styles",
        },
        "resultsScreen": Object {
          "bkgndImgDesktop": "",
          "bkgndImgMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg",
          "cta": "Read more from:",
          "ctaLink": "#MOVEMBER",
          "ctaTxt": "Thanks for voting!",
          "ctaUrl": "http://www.thesun.co.uk",
          "introTxt": "Here’s what everyone else thought, did you agree with the others?",
          "title": "The Results",
        },
        "teaserScreen": Object {
          "copy": "copy",
          "cta": "cta",
          "imageDesktop": "http://www.thesun.co.uk/imageDesktop.jpg",
          "imageMobile": "http://www.thesun.co.uk/imageMobile.jpg",
          "title": "title",
        },
        "voteScreen": Object {
          "bkgndImgDesktop": "desktop.jpg",
          "bkgndImgMobile": "handlebar-mobile.jpg",
          "cta": "Read more from:",
          "ctaLink": "#MOVEMBER",
          "ctaTxt": "Tell us what you think",
          "ctaUrl": "http://www.thesun.co.uk",
          "introTxt": "Which do you think looked best? Vote on your favourite moustache",
          "title": "Have your say",
        },
      }
    }
    id="card"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "backgroundColour": "#ffffff",
      "theme": Object {
        "primary": "#eb1701",
        "secondary": "#a5171c",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "backgroundColour": "#ffffff",
        "theme": Object {
          "primary": "#eb1701",
          "secondary": "#a5171c",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<PollContainer /> renders correctly when not errored 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PollContainer
    config={
      Object {
        "theme": Object {
          "primary": "#000000",
          "secondary": "#ffffff",
        },
      }
    }
    content={
      Object {
        "cards": Array [
          Object {
            "copy": "Looks a bit more Bruce Forsyth than Tom Selleck...",
            "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png",
            "id": "chevron",
            "imageDesktop": "",
            "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg",
            "title": "The Chevron",
          },
          Object {
            "copy": "Get a grip!",
            "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png",
            "id": "handlebar",
            "imageDesktop": "",
            "imageMobile": "",
            "title": "The Handlebar",
          },
          Object {
            "copy": "Work it out with this",
            "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png",
            "id": "pencil",
            "imageDesktop": "",
            "imageMobile": "",
            "title": "The Pencil",
          },
        ],
        "introScreen": Object {
          "cta": "Read more from:",
          "ctaLink": "#MOVEMBER",
          "ctaUrl": "http://www.thesun.co.uk",
          "imageDesktop": "",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png",
          "introText": Array [
            "Jeremy Kyle needs help picking his moustache style.",
            "Flick through the gallery of images and vote on your favourite at the end.",
          ],
          "title": "Jeremy Styles",
        },
        "resultsScreen": Object {
          "bkgndImgDesktop": "",
          "bkgndImgMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg",
          "cta": "Read more from:",
          "ctaLink": "#MOVEMBER",
          "ctaTxt": "Thanks for voting!",
          "ctaUrl": "http://www.thesun.co.uk",
          "introTxt": "Here’s what everyone else thought, did you agree with the others?",
          "title": "The Results",
        },
        "teaserScreen": Object {
          "copy": "copy",
          "cta": "cta",
          "imageDesktop": "http://www.thesun.co.uk/imageDesktop.jpg",
          "imageMobile": "http://www.thesun.co.uk/imageMobile.jpg",
          "title": "title",
        },
        "voteScreen": Object {
          "bkgndImgDesktop": "desktop.jpg",
          "bkgndImgMobile": "handlebar-mobile.jpg",
          "cta": "Read more from:",
          "ctaLink": "#MOVEMBER",
          "ctaTxt": "Tell us what you think",
          "ctaUrl": "http://www.thesun.co.uk",
          "introTxt": "Which do you think looked best? Vote on your favourite moustache",
          "title": "Have your say",
        },
      }
    }
    id="card"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Unknown
            cta="Read more from:"
            ctaLink="#MOVEMBER"
            ctaUrl="http://www.thesun.co.uk"
            imageDesktop=""
            imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png"
            introText={
              Array [
                "Jeremy Kyle needs help picking his moustache style.",
                "Flick through the gallery of images and vote on your favourite at the end.",
              ]
            }
            title="Jeremy Styles"
          />
          <glamorous(glamorous(div))>
            <glamorous(div)
              disabled={false}
              onClick={[Function]}
            >
              <Icon
                iconColour="#ffffff"
                iconName="chevron-solid-right"
                iconSize={16}
              />
            </glamorous(div)>
          </glamorous(glamorous(div))>
        </div>,
        false,
        false,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              cta="Read more from:"
              ctaLink="#MOVEMBER"
              ctaUrl="http://www.thesun.co.uk"
              imageDesktop=""
              imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png"
              introText={
                Array [
                  "Jeremy Kyle needs help picking his moustache style.",
                  "Flick through the gallery of images and vote on your favourite at the end.",
                ]
              }
              title="Jeremy Styles"
            />,
            <glamorous(glamorous(div))>
              <glamorous(div)
                disabled={false}
                onClick={[Function]}
              >
                <Icon
                  iconColour="#ffffff"
                  iconName="chevron-solid-right"
                  iconSize={16}
                />
              </glamorous(div)>
            </glamorous(glamorous(div))>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "cta": "Read more from:",
              "ctaLink": "#MOVEMBER",
              "ctaUrl": "http://www.thesun.co.uk",
              "imageDesktop": "",
              "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png",
              "introText": Array [
                "Jeremy Kyle needs help picking his moustache style.",
                "Flick through the gallery of images and vote on your favourite at the end.",
              ],
              "title": "Jeremy Styles",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <glamorous(div)
                disabled={false}
                onClick={[Function]}
              >
                <Icon
                  iconColour="#ffffff"
                  iconName="chevron-solid-right"
                  iconSize={16}
                />
              </glamorous(div)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Icon
                  iconColour="#ffffff"
                  iconName="chevron-solid-right"
                  iconSize={16}
                />,
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "iconColour": "#ffffff",
                  "iconName": "chevron-solid-right",
                  "iconSize": 16,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      false,
      false,
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Unknown
              cta="Read more from:"
              ctaLink="#MOVEMBER"
              ctaUrl="http://www.thesun.co.uk"
              imageDesktop=""
              imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png"
              introText={
                Array [
                  "Jeremy Kyle needs help picking his moustache style.",
                  "Flick through the gallery of images and vote on your favourite at the end.",
                ]
              }
              title="Jeremy Styles"
            />
            <glamorous(glamorous(div))>
              <glamorous(div)
                disabled={false}
                onClick={[Function]}
              >
                <Icon
                  iconColour="#ffffff"
                  iconName="chevron-solid-right"
                  iconSize={16}
                />
              </glamorous(div)>
            </glamorous(glamorous(div))>
          </div>,
          false,
          false,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                cta="Read more from:"
                ctaLink="#MOVEMBER"
                ctaUrl="http://www.thesun.co.uk"
                imageDesktop=""
                imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png"
                introText={
                  Array [
                    "Jeremy Kyle needs help picking his moustache style.",
                    "Flick through the gallery of images and vote on your favourite at the end.",
                  ]
                }
                title="Jeremy Styles"
              />,
              <glamorous(glamorous(div))>
                <glamorous(div)
                  disabled={false}
                  onClick={[Function]}
                >
                  <Icon
                    iconColour="#ffffff"
                    iconName="chevron-solid-right"
                    iconSize={16}
                  />
                </glamorous(div)>
              </glamorous(glamorous(div))>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "cta": "Read more from:",
                "ctaLink": "#MOVEMBER",
                "ctaUrl": "http://www.thesun.co.uk",
                "imageDesktop": "",
                "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png",
                "introText": Array [
                  "Jeremy Kyle needs help picking his moustache style.",
                  "Flick through the gallery of images and vote on your favourite at the end.",
                ],
                "title": "Jeremy Styles",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <glamorous(div)
                  disabled={false}
                  onClick={[Function]}
                >
                  <Icon
                    iconColour="#ffffff"
                    iconName="chevron-solid-right"
                    iconSize={16}
                  />
                </glamorous(div)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Icon
                    iconColour="#ffffff"
                    iconName="chevron-solid-right"
                    iconSize={16}
                  />,
                  "disabled": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "iconColour": "#ffffff",
                    "iconName": "chevron-solid-right",
                    "iconSize": 16,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        false,
        false,
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
